Parsing `export default ...` peeks to not overconsume tokens

Attamusc opened the pull request:
Fixes https://code.google.com/p/esprima/issues/detail?id=631

Paired with @mikesherov on this today.

Updates to the tests in `harmonymoduletest` to account for token generation is forthcoming (Will squash commits once those are in).

mikesherov commented on esprima.js:
```diff
@@ -3669,17 +3669,11 @@
 
     function parseExportDeclaration() {
         var backtrackToken, id, declaration = null,
```
isn't `bracktrackToken` no longer used?

Attamusc replied:
Yup, not used at all.

mikesherov commented on esprima.js:
```diff
@@ -3669,17 +3669,11 @@
 
     function parseExportDeclaration() {
         var backtrackToken, id, declaration = null,
+            peekedToken, sourceElement,
```
maybe we should rename `peekedToken` to `possibleIdentifierToken`?

Attamusc replied:
I like that better, since now it's just used to check if it's an identifier and not for potential rewinding.

michaelficarra commented:
Please squash.

mikesherov commented:
@michaelficarra yeah, we'll squash once tests land at the end. Thanks!

Attamusc commented:
@michaelficarra Sure thing! Should have mentioned I'd do that at in the description.

Attamusc commented:
Test changes are in. Since we modified code that touches the `default` branch of `parseExportDeclaration`, I added tokens to all `export default` tests.

mikesherov commented:
@jeffmo or @ariya can someone please review this when you have some free time?

jeffmo commented:
I'll take a look

-Jeff

On 2/6/15 9:48 AM, Mike Sherov wrote:
>
> @jeffmo <https://github.com/jeffmo> or @ariya 
> <https://github.com/ariya> can someone please review this when you 
> have some free time?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/jquery/esprima/pull/310#issuecomment-73246993>.
>

jeffmo commented on test/harmonymodulestest.js:
```diff
@@ -34,66 +34,203 @@ var modulesTestFixture = {
-                value: 42,
-                raw: '42',
-                range: [15, 17],
+            type: 'Program',
```
Any particular reason you're starting at the root in these tests? It seems the test runner will allow it, but the rest of the tests in this fixture only include the first statement in the program body...so unless there's a good reason it'd be nice to retain consistency.

jeffmo replied:
Oh, I missed your earlier comment here: https://github.com/jquery/esprima/pull/310#issuecomment-73166939

jeffmo commented:
This looks great, thanks for the PR!

Attamusc commented:
@jeffmo Ya the inconsistency in that file now is kind of a bummer, but I plan on adding tokens to the rest of the cases in `harmonymoduletest` later in another PR to make sure no other issues like this pop up.
